src/libsystemd/sd-daemon/sd-daemon.c

```diff
@@ -551,6 +552,77 @@ _public_ int sd_pid_notify_with_fds(
         return r;
 }
 
+_public_ int sd_notify_barrier(int timeout) {
```

Comment:
we generally unify using 64bit unsigned µs values for times. for internal code that means `usec_t`, and for externally visible APIs that means `uint64_t`

Reply:
i'd probably use the suffix `_sync()` for this, rather than `_barrier()`

Reply:
I have switched to uint64_t but time is still relative. About the naming, I think sd_notify_sync sounds more like a synchronous version of sd_notify, but I am not opposed to renaming things if you feel barrier as a name isn't appropriate.

Reply:
hmm, but it *is* a synchronous version of sd_notify(), no?

Reply:
hmm, actually i see what you mean. but then, gcc's memory barrier intrinsic is actually called `__sync_synchronize()` ;-)

but i see your point, let's call this barrier